components/session-based-conditional-auth-functions/src/main/java/org/wso2/carbon/identity/extensions/authenticator/conditional/auth/functions/function/IsWithinSessionLimitFunction.java

```diff
@@ -0,0 +1,165 @@
+            }
+            if (sessionCount < sessionLimit) {
+                state = true;
+            }
+        } catch (FrameworkException e) {
+            throw new AuthenticationFailedException("Problem occurred in session data retrieving", e);
+        } catch (NumberFormatException e) {
+            throw new AuthenticationFailedException("Failed to retrieve session count from response", e);
+        }
+        return state;
+    }
+
+    /**
+     * Method for retrieving user defined maximum session limit from parameter map
+     *
+     * @param map parameter map passed from JS
+     * @return inter indicating the maximum session Limit
+     */
+    private int getSessionLimitFromMap(Map<String, String> map) {
+
+        return parseInt(map.get(SessionValidationConstants.SESSION_LIMIT_TAG));
+    }
+
+    /**
+     * Method to retrieve active session count for the given authenticated user
+     *
+     * @param authenticatedUser Authenticated user object
+     * @return current active session count
+     * @throws FrameworkException When the REST response is not in 200 state or failed to read REST response
+     */
+    private int getActiveSessionCount(AuthenticatedUser authenticatedUser) throws FrameworkException {
+
+        int sessionCount;
+        JSONObject paramMap = new JSONObject();
+        paramMap.put(SessionValidationConstants.TABLE_NAME_TAG,
+                SessionValidationConstants.ACTIVE_SESSION_TABLE_NAME);
+        paramMap.put(SessionValidationConstants.QUERY_TAG,
+                SessionValidationUtil.getQuery(authenticatedUser.getTenantDomain(),
+                        authenticatedUser.getUserName(),
+                        authenticatedUser.getUserStoreDomain()));
+        if (log.isDebugEnabled()) {
+            log.debug("JSON payload for retrieving data :" + paramMap.toString());
+        }
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        StringEntity entity = new StringEntity(paramMap.toString(), ContentType.APPLICATION_JSON);
+        HttpClient httpClient = httpClientBuilder.build();
+        HttpPost request = new HttpPost(IdentityUtil.getProperty(SessionValidationConstants
+                .TABLE_SEARCH_COUNT_CONFIG_NAMES));
+
+        request = SessionValidationUtil.setAuthorizationHeader(request,
+                IdentityUtil.getProperty(SessionValidationConstants.USERNAME_CONFIG_NAME),
+                IdentityUtil.getProperty(SessionValidationConstants.PASSWORD_CONFIG_NAME));
+        request.addHeader(SessionValidationConstants.CONTENT_TYPE_TAG, "application/json");
+        request.setEntity(entity);
+        try {
+            HttpResponse response = httpClient.execute(request);
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(
```

Comment:
It would be safer to close your stream using a try..finally block.

Reply:
I was using try, final method. I was suggested to use this new try-with-resources method(https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)  during the code review